week7/answers.sql

```diff
@@ -1,3 +1,183 @@
 USE destruction;
+
+CREATE TABLE players (
+    player_id INT UNSIGNED PRIMARY KEY,
+    first_name VARCHAR(255),
+    last_name VARCHAR(255),
+    email VARCHAR(255)
+);
+CREATE TABLE characters (
+    character_id INT UNSIGNED PRIMARY KEY,
+    player_id INT UNSIGNED,
+    name VARCHAR(255),
+    level INT,
+    FOREIGN KEY (player_id) REFERENCES players(player_id)
+);
+CREATE TABLE winners (
+    character_id INT UNSIGNED PRIMARY KEY,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id)
+);
+CREATE TABLE character_stats (
+    character_id INT UNSIGNED PRIMARY KEY,
+    health INT,
+    armor INT,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id)
+);
+CREATE TABLE teams (
+    team_id INT UNSIGNED PRIMARY KEY,
+    name VARCHAR(255)
+);
+CREATE TABLE team_members (
+    team_member_id INT UNSIGNED PRIMARY KEY AUTO_INCREMENT,
+    team_id INT UNSIGNED,
+    character_id INT UNSIGNED,
+    FOREIGN KEY (team_id) REFERENCES teams(team_id),
+    FOREIGN KEY (character_id) REFERENCES characters(character_id)
+);
+CREATE TABLE items (
+    item_id INT UNSIGNED PRIMARY KEY,
+    name VARCHAR(255),
+    armor INT,
+    damage INT
+);
+CREATE TABLE inventory (
+    inventory_id INT UNSIGNED PRIMARY KEY,
+    character_id INT UNSIGNED,
+    item_id INT UNSIGNED,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id),
+    FOREIGN KEY (item_id) REFERENCES items(item_id)
+);
+CREATE TABLE equipped (
+    equipped_id INT UNSIGNED PRIMARY KEY,
+    character_id INT UNSIGNED,
+    item_id INT UNSIGNED,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id),
+    FOREIGN KEY (item_id) REFERENCES items(item_id)
+);
+
+CREATE VIEW character_items AS
+SELECT ch.character_id, ch.name AS character_name, i.name AS item_name, i.armor, i.damage
+FROM characters ch
+JOIN inventory inv ON ch.character_id = inv.character_id
+JOIN items i ON inv.item_id = i.item_id
+UNION
+SELECT ch.character_id, ch.name AS character_name, i.name AS item_name, i.armor, i.damage
+FROM characters ch
+JOIN equipped eq ON ch.character_id = eq.character_id
+JOIN items i ON eq.item_id = i.item_id;
+
+CREATE VIEW team_items AS
+SELECT tm.team_id, t.name AS team_name, i.name AS item_name, i.armor, i.damage
+FROM teams t
+JOIN team_members tm ON t.team_id = tm.team_id
+JOIN characters ch ON tm.character_id = ch.character_id
+JOIN inventory inv ON ch.character_id = inv.character_id
+JOIN items i ON inv.item_id = i.item_id
+UNION
+SELECT tm.team_id, t.name AS team_name, i.name AS item_name, i.armor, i.damage
+FROM teams t
+JOIN team_members tm ON t.team_id = tm.team_id
+JOIN characters ch ON tm.character_id = ch.character_id
+JOIN equipped eq ON ch.character_id = eq.character_id
+JOIN items i ON eq.item_id = i.item_id;
+
+DELIMITER //
+
+CREATE FUNCTION armor_total(character_id INT) RETURNS INT
+READS SQL DATA
+BEGIN
+    DECLARE total_armor INT;
+    
+    SELECT COALESCE(SUM(cs.armor), 0) + COALESCE(SUM(i.armor), 0) AS total
```

Comment:
I think you'll find it easier (and the math will work out if you make this change) if you do this in two unrelated queries instead of a single SELECT (and you won't get any NULL values). This isn't far off, but looking at the automated tests, the math doesn't quite work out.